performance/query_processing.rst

```diff
@@ -34,7 +34,9 @@ Distributed Query Executor
 
 Citus’s distributed executors run distributed query plans and handle failures that occur during query execution. The executors connect to the workers, send the assigned tasks to them and oversee their execution. If the executor cannot assign a task to the designated worker or if a task execution fails, then the executor dynamically re-assigns the task to replicas on other workers. The executor processes only the failed query sub-tree, and not the entire query while handling failures.
 
-Citus has two executor types - real time and task tracker. The former is useful for handling simple key-value lookups and INSERT, UPDATE, and DELETE queries, while the task tracker is better suited for larger SELECT queries.
+Citus has three basic executor types: real time, router, and task tracker. It can use more than one at once on a single query, assigning different executors to different subqueries/CTEs as needed to support the SQL functionality. This process is recursive: if Citus cannot determine how to run a subquery then it examines sub-subqueries.
```

Comment:
Do we want to note somewhere that all of this is handled by Citus automatically? (i.e., the executor choice is "Dynamic"), and the user needs not take any action.
We can also show an `EXPLAIN` example to allow users to determine which executor a query is using. E.g., 
"if you want to know which executor a query is using to debug performance issues, you can check our explain plan".